src/Lykke.Service.PayInternal/Controllers/TransferRequestController.cs

```diff
@@ -1,11 +1,244 @@
 using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.SwaggerGen;
 
 namespace Lykke.Service.PayInternal.Controllers
 {
-    public class TransferRequestController: Controller
+    [Route("api")]
+    public class TransferRequestController : Controller
     {
+        private readonly ITransferRequestService _transferRequestService;
+        public TransferRequestController(ITransferRequestService transferRequestService)
+        {
+            _transferRequestService = transferRequestService;
+        }
+
+        /// <summary>
+        /// Request to transfer all money.
+        /// </summary>
+        /// <param name="merchantId">Merchant Id.</param>
+        /// <param name="destinationAddress">Destination Address.</param>
+        /// <returns>The Transfer Info.</returns>
+        /// <response code="200">The Transfer Info.</response>
+        /// <response code="400">Invalid model.</response>
+        [HttpPost]
+        [Route("merchants/{merchantId}/transferAll/{destinationAddress}")]
+        [SwaggerOperation("TransferRequestAll")]
+        [ProducesResponseType(typeof(PaymentRequestModel), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> TransferRequestAllAsync(string merchantId, string destinationAddress)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(new ErrorResponse().AddErrors(ModelState));
+
+            if (string.IsNullOrEmpty(merchantId))
+                return BadRequest(ErrorResponse.Create("Merchant Id doesn't exist"));
+
+            if (string.IsNullOrEmpty(destinationAddress))
+                return BadRequest(ErrorResponse.Create("Destination Address doesn't exist"));
+
+            return Ok(await _transferRequestService.CreateTransferAsync(new TransferRequestModel
+            {
+                DestinationAddress = destinationAddress,
+                MerchantId = merchantId
+            }));
+        }
+
+        /// <summary>
+        /// Request to transfer specify amount.
+        /// </summary>
+        /// <param name="merchantId">Merchant Id.</param>
+        /// <param name="destinationAddress">Destination Address.</param>
+        /// <param name="amount">Amount to send</param>
+        /// <returns>The Transfer Info.</returns>
+        /// <response code="200">The Transfer Info.</response>
+        /// <response code="400">Invalid model.</response>
+        [HttpPost]
+        [Route("merchants/{merchantId}/transferAll/{destinationAddress}/amount/{amount}")]
+        [SwaggerOperation("TransferRequestAmountAll")]
+        [ProducesResponseType(typeof(PaymentRequestModel), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ErrorResponse), (int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> TransferRequestAmountAsync(string merchantId, string destinationAddress, string amount)
```

Comment:
Use model or query parameters for address and amount. Why amount is string?

Reply:
amount is a string because if it double, but query contains NAN value API will throw 500 error instead of custom